### ISSUE TITLE
Support for dynamically updating currency and locale of MoneyInput

### ISSUE THREAD

**Issue by AlexanderBV** [pelmered/filament-money-field#42]
**Describe the bug**
I'm encountering an error when trying to dynamically set the currency and locale for the `MoneyInput` component in Filament forms. The error occurs specifically when trying to use a closure for the `currency` property.

**Error Message**
Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization

The error is triggered by this line:
```php
->currency(fn ($get) => $get('unit_price_currency'))

Steps To Reproduce:

```php
use Filament\Forms\Components\Grid;
use Filament\Forms\Components\Hidden;
use Filament\Forms\Components\Select;
use Pelmered\FilamentMoneyField\Forms\Components\MoneyInput;
use Filament\Forms\Form;
use Filament\Forms\Get;
use Filament\Forms\Set;
use App\Models\Catalog;
use App\Enums\RateType;

public function form(Form $form): Form
{
    return $form
        ->schema([
            Select::make('currency_id')
                ->label('Moneda')
                ->relationship('currency', 'name')
                ->searchable(['code', 'name'])
                ->preload()
                ->reactive()
                ->required()
                ->afterStateUpdated(function (Set $set, Get $get) {
                    $currency = Catalog::find($get('currency_id'));
                    $set('unit_price_currency', $currency->code);
                    $set('unit_price_locale', $currency->properties['locale'] ?? 'es_PE');
                })
                ->afterStateHydrated(function (Set $set, Get $get, $state) {
                    $currency = Catalog::find($state);
                    $set('unit_price_currency', $currency->code);
                    $set('unit_price_locale', $currency->properties['locale'] ?? 'es_PE');
                }),
            Hidden::make('unit_price_currency'),
            Hidden::make('unit_price_locale'),
            MoneyInput::make('unit_price')
                ->label('Precio')
                ->currency(fn ($get) => $get('unit_price_currency'))
                ->locale(fn ($get) => $get('unit_price_locale'))
        ]);
}
```

Changes have in global config:

// .env file settings
MONEY_DEFAULT_LOCALE=es_PE
MONEY_DEFAULT_CURRENCY=PEN

// Configuration published with
php artisan vendor:publish --provider="Pelmered\FilamentMoneyField\FilamentMoneyFieldServiceProvider" --tag="config"

Expected behavior and actual behavior
I expected the MoneyInput component to correctly use the dynamically set currency and locale values without causing initialization errors. Instead, it throws an error indicating that the property $container must not be accessed before initialization.

Screenshots
![Screenshot from 2024-06-26 11-31-49](https://github.com/pelmered/filament-money-field/assets/20906718/7d312658-3d56-4e39-8f01-102a86706384)

Additional context
I believe the issue might be related to the initialization of the $container property, which seems to be accessed before it is fully initialized. Any insights or solutions to this problem would be greatly appreciated. Thank you for your assistance!

Version of:
This package: Latest
Filament: 3.2
PHP: 8.2
Laravel: 10.10

**Comment by pelmered** [comment]
Thank you for the report! I will investigate this today.

**Comment by pelmered** [comment]
I don't believe this error has anything to with this plugin. It is how Filament works. 

I think you have to do something like this:
```
                MoneyInput::make('price')
                          ->label('Precio')
                          ->afterStateUpdated(function (MoneyInput $component, bool $state) {
                              $component->currency(fn($get) => $get('unit_price_currency') ?? 'USD')
                                        ->locale(fn($get) => $get('unit_price_locale') ?? 'es_PE');
                          })
```

**Comment by pelmered** [comment]
I have now tested some, and I'm not sure how well it works to change the field dynamically like this. This is something I will have to look into in the future. For now I would probably just use a currency select and a normal input without localized format. For displaying it should work with the `MoneyColumn` and the `MoneyEntry`.

If you find a way to support this, please let me know.

**Comment by bilogic** [comment]
I think this issue will be addressed if we figure out how prices should be persisted. My view of s `price` is that it should always consist of a `currency` and `value`.

Is there a recommended way? Especially if there are multiple prices in record.

**Comment by pelmered** [comment]
> I think this issue will be addressed if we figure out how prices should be persisted. My view of s `price` is that it should always consist of a `currency` and `value`.
> 
> Is there a recommended way? Especially if there are multiple prices in record.

Yes, I would agree with that! 
I'm not sure how that would be implemented and stored in the database. I guess it has to be separate columns. This is something I would like to work on for a 2.0 release. I you have ideas I'm happy for feedback and ideas on how to do this. 

I guess the best way to do this is a integer/decimal column for amount and one char(3) for currency. We should add a migration type for this like for example the morph type that created two related columns.

**Comment by ghsgabriel** [comment]
Hello,

I have been facing the same issue. This was my alternative solution:

```php
  Select::make('country')
      ->live()
      ->afterStateUpdated(fn (Get $get, Select $component) => $component
          ->getContainer()
          ->getComponent('consuption_limit')
          ->currency(fn($get) =>  Country::find( $get('country') )->currency)
          ->locale(fn($get) => Country::find( $get('country') )->locale)
      )
      ->relationship('country', 'name'),
  MoneyInput::make('consuption_limit')
      ->hidden(fn (Get $get): bool => ! $get('country'))
      ->key('consuption_limit')
```

I'm not sure how the official plugin implementation could be, but I hope this can give you an idea, or at least help someone facing the same difficulty.

Regarding how prices should be persisted, I think it varies from application to application. In my application, I have 2 columns, one for currency and one for value.

**Comment by pelmered** [comment]
This is fixed in 2.0. It will be released in beta this week!
